### PR TITLE
WIP: Add Press Kit

### PR DESCRIPTION
**Pull request by TobiGr**
That PR is mostly for you to have a look at the content.

These things have to be done before merging:

- [x] Fix the press kit search (lunrjs/press_searchData.js)

- [ ] Clarify some creators of our media.
a) @bagelcake You created our the feature graphic. How should I mention you? Atm it's done via you GitHub account. And how do we license it?

- [ ] Clean up the whole code

- [x] Generate new screenshot series with latest release.

- [x] Rework Contact page 
@theScrabi @TheAssassin What do you think should be removed?

- [x] Add announcements (atm there are only placeholders) and improve the design for this page.
- [x] Make footer sticky.

- [x] Use Jekyll to 
a) reduce duplicate code.
b) make applying changes easier.
c) get a good search with lunr.js.
Maybe we then should move the press kit to another repository.

- [x] Add 404 error page

Ideas for a later point of time:
Either create a nice Jekyll/PHP (or whatever) plugin to generate press kit as PDF out of our website or generate a nice template for TeX/LaTeX, Word, GIMP/Photoshop etc.

### CODE REVIEW

**Comment by TheAssassin** [comment]
Re PDF generation: we can just use https://wkhtmltopdf.org/ for that purpose. It takes the `print` media tags into account, so we can add a few print specific lines of CSS code to improve the look of the press kit PDF. We can even structure the HTML to fit PDF pages properly relatively easy, using e.g., https://davidwalsh.name/css-page-breaks.

**Comment by TobiGr** [comment]
With adding Jekyll, we should consider to put the press kit into another repository. The search isn't that perfect, but I'll improve it with the next commits. Most important is the content of our pages. That's why please take a look at it and tell me what texts need improvements.
@theScrabi @TheAssassin For the announcements page I need at least two or three good articles. A press kit without any press information is pointless.

File: _includes/head.html [hunk 1]
```diff
@@ -31,4 +31,5 @@
     <link rel="stylesheet" type="text/css" href="{{ site.baseurl }}/css/press.css" />
 
 </head>
-<body>
\ No newline at end of file
+<body>
+<div id="page">
```
Comment (TheAssassin) [inline _includes/head.html]: @TobiGr you should change your editor's settings to append a trailing newline.

File: css/style.css [hunk 2]
```diff
@@ -694,4 +697,27 @@ body.modal-open {
...
+
+.height-full-parent {
+    display: -ms-flex; display: -webkit-flex; display: flex;
+}
+.height-full-child {
+    flex: 1;
+}
+.site-content {
+    flex: 1 0 auto;
 }
```
Comment (TheAssassin) [inline css/style.css]: @TobiGr you should change your editor's settings to append a trailing newline.

**Comment by TobiGr** [comment]
I added some more texts. Do you think they are too promotional and not neutral enough?

**Comment by TobiGr** [comment]
> Re PDF generation: we can just use https://wkhtmltopdf.org/ for that purpose. It takes the print media tags into account, so we can add a few print specific lines of CSS code to improve the look of the press kit PDF. We can even structure the HTML to fit PDF pages properly relatively easy, using e.g., https://davidwalsh.name/css-page-breaks.

@TheAssassin That works! Thanks. I'll work out a nice layout for the PDFs. 
@theScrabi I'd like to publish the press kit until 27th December. That's why please review all texts so I can  take care off all problems and fix them.

**Comment by TheAssassin** [comment]
@TobiGr I guess a press kit has to be promotional. This is nothing that requires neutrality.

Good marketing is something promotional that doesn't sound too promotional, I'd say.

**Comment by TobiGr** [comment]
> I guess a press kit has to be promotional. This is nothing that requires neutrality.

Press kits should help editors of newspapers, magazine etc. to quickly write their articles. Many of them do not have enough time to write completely new articles. Therefore, it should be possible for them to copy texts from press kits directly (or with only minor changes) into the article. Thus the text needs a kind of journalist quality - i.a. neutrality -  and the appropriate easy-to-understand language.
At the same time they have to show the readers the interesting things and convince them to purchase / promote the items. 
When you find any sentences which are too promotional and not neutral enough, we should improve them.

**Comment by theScrabi** [comment]
> Thus the text needs a kind of journalist quality

I need someone to help me on this, since this is definitely not one of my best abilitys. @TheAssassin can we make this together?

**Comment by TheAssassin** [comment]
@theScrabi yes, I've sent you the pad link already. Just start writing something, I can finish it then.

**Comment by TobiGr** [comment]
@TheAssassin @theScrabi Can I take the blog posts and add them to the announcements page? If yes, I think we are almost ready to go and only the contact page is left for a quick review.

**Comment by TheAssassin** [comment]
Wouldn't it make more sense to just link to the blog to avoid having to maintain the same text in multiple places?

**Comment by TobiGr** [comment]
When we make an announcement - and that is not so often - we generally do not change it. That's why maintaining shouldn't be a problem. And linking to a blog with a tag filter does not look that professional to me.

**Comment by TheAssassin** [comment]
> we generally do not change it

May I ask why you believe so? There might always be a wrong fact, a typo, etc. Sure, the overall wording *should* not change, but I can imagine a lot of cases in which the text could need some adjustments.

**Comment by TobiGr** [comment]
> I can imagine a lot of cases in which the text could need some adjustments

That's right, but if you change one text, it shouldn't be that much work to copy the code and paste it into the other page.
I want people who visit the press kit should be as comfortable as possible. That's why I'd like to have all information at one place for them. And another place with different information for the "normal" audience.

**Comment by TheAssassin** [comment]
> it shouldn't be that much work

No, but it has to be done, and having different texts on the pages is pretty annoying. What if someone forgets to do so?

Since both pages are Jekyll based, it might make sense to combine both and move the blog into this repository, too. The repository title "BlogGenerator" is a bit misleading anyway, initially I planned to rename it, but I guess to save some bloat, we could just move the blog infrastructure here, too. Then, all pages can be generated from this repository, and the blog texts can be rendered in the press kit, too.

How much work would it be to do so? If it is too much effort, then I think we can work with the current situation, too, of course.

**Comment by TobiGr** [comment]
> How much work would it be to do so? If it is too much effort, then I think we can work with the current situation, too, of course.

It wouldn't be that hard to implement. The only important thing we have to talk about first, is the structure. Since some files are equal (Bootstrap, JQuery, Logo), there are also some files with the same name which have different source code (style.css, search, etc.). We could rename/prefix them. Also the press kit has many images. I'd prefer to create a subdirectory for it.

**Comment by TheAssassin** [comment]
Feel free to improve the directory structure. As the main developer, you are invited to make any change that improves the repository, of course. We really appreciate your work.

**Comment by TobiGr** [comment]
@TheAssassin Adding the press kit to the blog repository was a bit trickier than I thought because we can't use any directories besides `press` and `blog`. Bit everything works fine now except for the search. I can take a look at it tomorrow.

**Comment by TobiGr** [comment]
@TheAssassin I also added to 404 error pages: `blog/404.html` and `press/404.html`. I don't know what server you use, so you'll have to add `.htaccess` or `nginx.conf` files to both directories.

**Comment by TheAssassin** [comment]
> @TheAssassin I also added to 404 error pages: blog/404.html and press/404.html. I don't know what server you use, so you'll have to add .htaccess or nginx.conf files to both directories.

That's not an issue at all, due to the Docker deployment, we can just add a working config file to the container.

**Comment by TobiGr** [comment]
Perfect.

**Comment by TobiGr** [comment]
@TheAssassin How do we want to call the blog and press repo now?

**Comment by TheAssassin** [comment]
Actually, I thought you were merging everything into this repository? I just saw now you closed this PR. Is it possible to put it all into one location? That seems like the most viable solution, and makes deployment a bit easier. If they can't be mixed, we can also move the blog and presskit into one directory, and the static website into another one, and I'll put everything together via docker-compose.

**Comment by TobiGr** [comment]
Oh. That was a typical misunderstanding. Yes it is possible and easier to put everything into this repository. I could have save some time if I understood you correctly 😂

**Comment by TheAssassin** [comment]
@TobiGr yeah, that's a problem I regularly experience with others in free software projects. Especially in more international ones, where the timezones differ highly. I'm sorry about it.

**Comment by TobiGr** [comment]
@TheAssassin You don't have to be sorry. That's the disadvantage of using text chats. Sometimes nothing beats a personal conversation.

**Comment by TheAssassin** [comment]
@TobiGr can you estimate how long you will need until we can merge the blog and presskit into the website repository? I don't want to hustle you, but we want to release something as soon as possible. Otherwise, the press release regarding the talk on 34C3 can't be shared in time on the social media platforms. If you could use some help, feel free to post a TODO, or create issues, which we can help you with.

**Comment by TobiGr** [comment]
@TheAssassin If you want, we can squash and merge it now. There is one Jekyll bug, which leads to wrong search results for the press kit (`lunrjs/press_searchData.js`). `various.html` and `screenshots.html` are not parsed correctly for the search (the normal sites are parsed as expected). I don't have time to take a look at it now. I didn't check the file encoding, but it should be UTF-8. If you can't find a quick solution for this problem, I'd open a ticket in the Jekyll forum.

**Comment by TobiGr** [comment]
@TheAssassin I couldn't fix the original bug, but I have a hotfix so the search works as expected. 
What do we with the blog sidebar?

**Comment by TheAssassin** [comment]
I wouldn't say the sidebar has any priority, we can fix it eventually. If you want to, we can hardcode the current blog posts like you did before, and leave a TODO there (maybe even open an issue in the repository). What do you think about this idea?

Is it ready for merge otherwise?

**Comment by TobiGr** [comment]
@TheAssassin What post should be pinned there?

> Is it ready for merge otherwise?

Yes it is.

**Comment by TheAssassin** [comment]
@TobiGr I'd say, why not pin all the posts? We don't have many, and I think it might look more "interesting" with as many entries in there as possible at the moment. But if you disagree about some posts, feel free to pin only those which you want to pin.

I'd add `pinned` as a category to all those posts, and hardcode their URLs in the sidebar. That allows for replacing the hardcoded parts with the sidebar generator once it is ready, preserving the sidebar structure, and gives authors an idea of what it will work like.

By the way, I noticed that after renaming the posts to `.md` instead of `.html` (they're actually Markdown), the `.md` suffix shows up in the URLs. Feel free to revert this change, `.html` looks better. I can't find an option in Jekyll to automatically change the suffix of the output files.

**Comment by TobiGr** [comment]
> By the way, I noticed that after renaming the posts to .md instead of .html (they're actually Markdown), the .md suffix shows up in the URLs.

I can't reproduce this.

**Comment by TobiGr** [comment]
@TheAssassin @theScrabi I think we are ready to go.

**Comment by TheAssassin** [comment]
Feel free to merge. I'll work out a deployment strategy afterwards.

**Comment by TobiGr** [comment]
Finally everything is done.

**Comment by TheAssassin** [comment]
Thank you very much, @TobiGr, your work is invaluable for the project!

**Pull request merged**